pandas/core/arraylike.py

```diff
@@ -149,6 +149,85 @@ def __rpow__(self, other):
         return self._arith_method(other, roperator.rpow)
 
 
+# -----------------------------------------------------------------------------
+# Helpers to implement __array_ufunc__
+
+
+def _is_aligned(frame, other):
+    """
+    Helper to check if a DataFrame is aligned with another DataFrame or Series.
+    """
+    from pandas.core.frame import DataFrame
```

Comment:
might as well just import from pandas here, this is *only* the import if you can import at the top of the file (not sure if you can), also maybe can use ABCDataFrame

Reply:
``pandas.core.frame.py`` import from this file, so I don't think I can move the import to the top of the file

Reply:
i get that you cannot put the import at the top. However when *inside* the function the style is to 
```from pandas import DataFrame```

Reply:
OK, changed the imports